org.uqbar.project.wollok.tests/src/org/uqbar/project/wollok/tests/interpreter/mixins/MixinsTestCase.xtend

```diff
@@ -0,0 +1,369 @@
+			var fliedMeters = 0
+			method fly(meters) {
+				this.reduceEnergy(meters)
+				fliedMeters += meters
+			}
+			method fliedMeters() = fliedMeters
+			
+			method reduceEnergy(meters)
+		}
+		
+		class BirdWithEnergyThatFlies mixed with Energy, Flying {}
+		
+		// order doesn't matter
+		
+		class BirdWithThatFliesWithEnergy mixed with Flying, Energy {}
+		
+		program t {
+				val b = new BirdWithEnergyThatFlies()
+				b.fly(10)
+				assert.equals(90, b.energy())
+				assert.equals(10, b.fliedMeters())
+				
+				val b2 = new BirdWithThatFliesWithEnergy()
+				b2.fly(10)
+				assert.equals(90, b2.energy())
+				assert.equals(10, b2.fliedMeters())
+		}
+		'''.interpretPropagatingErrors
+	}
+	
+	@Test
+	def void mixinMethodCallsAMethodOnTheClass() {
+		'''
+		mixin FlyingShortcuts {
+			method fly100Meters() {
+				this.fly(100)
+			}
+			method fly(meters)
+		}
+		
+		class BirdWithFlyingShortCuts mixed with FlyingShortcuts {
+			var energy = 200
+			override method fly(meters) { energy -= meters }
+			method energy() = energy
+		}
+		
+		program t {
+			val b = new BirdWithFlyingShortCuts()
+			b.fly100Meters()
+			assert.equals(100, b.energy())
+		}
+		'''.interpretPropagatingErrors
+	}
+	
+	@Test
+	def void mixinMethodCallsAMethodOnTheClassThatIsInherited() {
+		'''
+		class Bird {
+			var energy = 200
+			method fly(meters) { energy -= meters }
+			method energy() = energy
+		}
+		
+		mixin FlyingShortcuts {
+			method fly100Meters() {
+				this.fly(100)
+			}
+			method fly(meters)
+		}
+		
+		class MockingBird inherits Bird mixed with FlyingShortcuts {
+		}
+		
+		program t {
+			val b = new MockingBird()
+			b.fly100Meters()
+			assert.equals(100, b.energy())
+		}
+		'''.interpretPropagatingErrors
+	}
+	
+	@Test
+	def void classOverridesMethodFromMixin() {
+		'''
+		mixin Energy {
+			var energy = 100
+			method reduceEnergy(amount) { energy -= amount }
+			method energy() = energy
+		}
+		
+		class Bird mixed with Energy {
+			override method reduceEnergy(amount) { 
+				// does nothing
+			}
+		}
+		
+		program t {
+			val b = new Bird()
+			b.reduceEnergy(100)
+			assert.equals(100, b.energy())
+		}
+		'''.interpretPropagatingErrors
+	}
+	
+	@Test
+	def void classOverridesMethodFromMixinAndCallSuper() {
+		'''
+		mixin Energy {
+			var energy = 100
+			method reduceEnergy(amount) { energy -= amount }
+			method energy() = energy
+		}
+		
+		class Bird mixed with Energy {
+			override method reduceEnergy(amount) { 
+				super(1)
+			}
+		}
+		
+		program t {
+			val b = new Bird()
+			b.reduceEnergy(100)
+			assert.equals(99, b.energy())
+		}
+		'''.interpretPropagatingErrors
+	}
+	
+	@Test
+	def void stackableMixinsPattern() {
+		'''
+		mixin M1 {
+			method doFoo(chain) { super(chain + " > M1") }
+		}
+		
+		mixin M2 {
+			method doFoo(chain) { super(chain + "> M2") }
+		}
+		
+		class C1 {
+			var foo = ""
+			method doFoo(chain) { foo = chain + " > C1" }
+			method foo() = foo
+		}
+		
+		class C2 inherits C1 mixed with M1, M2 {
```

Comment:
Here you use 'mixed with M1, M2' while before you have used 'mixed with M1 and M2', are both right?